experimental/vjn/yarpl/include/reactivestreams/subscriber.h

```diff
@@ -0,0 +1,94 @@
+#pragma once
+
+#include <exception>
+#include <memory>
+
+#include "type_traits.h"
+
+namespace reactivestreams {
+
+template <typename T>
+class Subscription;
+
+template <typename T>
+class Subscriber {
+ public:
+  virtual ~Subscriber() = default;
+
+  virtual void on_next(const T&) {}
+  virtual void on_next(T&& value) { on_next(value); }
```

Comment:
this overload seems unnecessary as rvalue reference will automatically bind to const lvalue reference...